gordon_janitor_gcp/plugins/authority.py

```diff
@@ -0,0 +1,185 @@
+        await authority.start()
+        msg = await rrset_channel.get()
+        print(msg)
+
+    loop = asyncio.get_event_loop()
+    loop.run_until_complete(run())
+    # prints: {'zone': 'example.com', 'resourceRecords': [...]}
+"""
+
+import asyncio
+import logging
+
+import aiohttp
+import zope.interface
+from gordon_janitor import interfaces
+
+from gordon_janitor_gcp import exceptions
+from gordon_janitor_gcp.clients import auth
+from gordon_janitor_gcp.clients import gce
+from gordon_janitor_gcp.clients import gcrm
+
+
+class GCEAuthorityBuilder:
+    """Class that helps create a properly configured GCEAuthority object.
+
+    Args:
+        config (dict): plugin-specific configuration.
+        rrset_channel (asyncio.Queue): channel to send resource record messages
+            to.
+    """
+    def __init__(self, config, rrset_channel, **kwargs):
+        self.config = config
+        self.rrset_channel = rrset_channel
+        self.kwargs = kwargs
+        self._loop = asyncio.get_event_loop()
+        self.session = aiohttp.ClientSession(loop=self._loop)
+
+    def _get_or_raise(self, key):
+        item = self.config.get(key)
+        if not item:
+            msg = f'gcp.gce configuration missing "{key}" setting'
+            logging.error(msg)
+            # Close the session if any required config is missing.
+            self.session.close()
+            raise exceptions.GCPConfigError(msg)
+        return item
+
+    def _get_crm_client(self, keyfile_path, scopes):
+        crm_auth = auth.GAuthClient(
+            keyfile_path, scopes=scopes, session=self.session)
+        return gcrm.GCRMClient(crm_auth, self.session)
+
+    def _get_gce_client(self, keyfile_path, scopes):
+        tag_blacklist = self.config.get('tag_blacklist', [])
+        _metadata_blacklist = self.config.get('metadata_blacklist', [])
+        metadata_blacklist = [dict([pair]) for pair in _metadata_blacklist]
+
+        gce_auth = auth.GAuthClient(keyfile_path, scopes=scopes,
+                                    session=self.session)
+        return gce.GCEClient(gce_auth, self.session,
+                             blacklisted_tags=tag_blacklist,
+                             blacklisted_metadata=metadata_blacklist)
+
+    def build_authority(self):
+        """Coordinate dependencies and create a GCEAuthority object.
+
+        Returns:
+            obj: A configured GCEAuthority object.
+        """
+        self._get_or_raise('zone')
+        keyfile_path = self._get_or_raise('keyfile')
+        scopes = self._get_or_raise('scopes')
+        crm_client = self._get_crm_client(keyfile_path, scopes)
+        gce_client = self._get_gce_client(keyfile_path, scopes)
+
+        return GCEAuthority(self.config, crm_client, gce_client,
+                            self.rrset_channel, **self.kwargs)
+
+
+@zope.interface.implementer(interfaces.IAuthority)
+class GCEAuthority:
+    """Gordon-Janitor plugin that gathers "state of truth" GCE instance data.
```

Comment:
Also should be in the imperative, e.g. "Get instance data from source of truth."